arraycontext/impl/pytato/compile.py

```diff
@@ -415,12 +415,23 @@ def _dag_to_transformed_pytato_prg(self, dict_of_named_arrays, *, prg_id=None):
                 prg_id, "pre_generate_loopy", pt_dict_of_named_arrays)
 
         with ProcessLogger(logger, f"generate_loopy for '{prg_id}'"):
+            import pyopencl as cl
+            dev = self.actx.context.devices[0]
+            target = None
+            if (dev.type & cl.device_type.GPU
+                    and cl.characterize.has_coarse_grain_buffer_svm(dev)):
+                limit = dev.max_parameter_size
+                # Leave some extra space since our sizes are estimates
+                target = lp.PyOpenCLTarget(limit_arg_size_nbytes=limit//2)
+
             pytato_program = pt.generate_loopy(
                     pt_dict_of_named_arrays,
                     options=lp.Options(
                         return_dict=True,
                         no_numpy=True),
-                    function_name=_prg_id_to_kernel_name(prg_id))
+                    function_name=_prg_id_to_kernel_name(prg_id),
```

Comment:
Factor the target construction into a memoized function and also use that in `freeze`?

Reply:
Done in Done in [1e54ce4](https://github.com/inducer/arraycontext/pull/189/commits/1e54ce49945684008ab48a1a7f2793fc5590a702)